interview-questions.md

```diff
@@ -7,44 +7,78 @@ Then, research the question to expand on your answer. Even if you feel you have
+  Create a table animals and has two column name and decription
+  -rails g resource
+  Create a model, controller, rspec file, and restful routes for the app
+  - rials g controller
+  Create a controller for the app which is coordinates the interaction between the user, the views, and the model.
+
 
   Researched answer:
+  rails g model Animal
+  A Rails Model is a Ruby class that can add database records (think of whole rows in an Excel table), find particular data you're looking for, update that data, or remove data. These common operations are referred to by the acronym CRUD--Create, Remove, Update, Destroy.
 
+  rials g controller
+  RSpec is a testing tool for Ruby, created for behavior-driven development (BDD). It is the most frequently used testing library for Ruby in production applications.
 
+  rials g controller
+  The Rails controller is the logical center of your application. It coordinates the interaction between the user, the views, and the model. The controller is also a home to a number of important ancillary services.
 
-4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
-action: "GET"    route: /students          
 
-action: "POST"   route: /students       
+4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
+
+action: "GET"    route: /students === index (to show all of students) students/index 
 
-action: "GET"    route: /students/new
+action: "POST"   route: /students === create (to creat a new student) stusdents/create
+ 
+action: "GET"    route: /students/new === new (return an HTML form for creating a new students)
 
-action: "GET"    route: /students/2  
+action: "GET"    route: /students/2 === show (to display specific student)
 
-action: "GET"    route: /students/2/edit    
+action: "GET"    route: /students/2/edit === edit (return an HTML form for editing a student)
 
-action: "PATCH"  route: /students/2      
+action: "PATCH"  route: /students/2 === update (update a specific student)
 
-action: "DELETE" route: /students/2      
+action: "DELETE" route: /students/2 === destroy (delete a specific student)
 
 
 
 5. As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+
+Create a table to keep track of 10 users and their tasks:
+I will have a table User and have a name, task, status, and deadline
+- name (it will have their names)
+- task (it will have name of their tasks)
+- status -- (it will have the status of task "in progress" or "completed")
+- deadline -- when is the deadline (it will have the deadline when we expect this project to be done)
+
```

Comment:
It seems like you were looking at this as a developer which is good. The question wanted a user perspective. So based on what you build as a developer. For instance,
As a user, I can see a list with column headers of a name, task, status, and deadline.
As a user, I can update the name in that list.
As a user, I can create a new task.